db/migrate/20250212192052_add_constraints_to_user_action_events.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class AddConstraintsToUserActionEvents < ActiveRecord::Migration[7.2]
+  def change
+    safety_assured do
```

Comment:
This might be unnecessary but do we want to add a `UserActionEvent.delete_all` just to be safe given the danger of adding a not-null constraint to an existing entry? I don't think there are any rows saved now but might be in the future and `UserActionEvents` will all be generated programmatically from the config so there's data to lose.